lib/packages/docutils/rstast.nim

```diff
@@ -77,6 +77,10 @@ type
 
   PRstNode* = ref RstNode    ## an RST node
   RstNodeSeq* = seq[PRstNode]
+  PFootnoteRefInfo* = ref object ## location inside a file, only for
```

Comment:
ref object seems like a false optimization; the cost of allocs will dwarf any savings here (not that those savings should matter much, but not using ref seems simpler here anyways)

I'm betting the total number of `RstNode` nodes generated is itself dwarfed by total number of PNode generated during nim doc anyways.

Reply:
may be. But araq previously complained about similar case.

Reply:
where? in fact https://github.com/nim-lang/Nim/pull/18383#discussion_r663351147 tells the same thing: 
> Why would it ever be a ref TLineInfo? Don't allocate these small objects on the heap.

but this can be addressed in followup PR, not a blocker

Reply:
In another PR. It was in a similar situation (when 1 extra field was proposed for addition) a few months ago.

Reply:
actually saving `order` field is not necessary for `rnFootnoteRef`, instead I added 2 counters to `RstSharedState` that are processed in `resolveSubs`.

So the problem is gone without need to sacrifice any memory consumption.

@timotheecour , see the latest commit.

Reply:
ok!